### PR TITLE
build: Migrate pause image building to k/release

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

- build: Migrate pause image building to k/release (accompanying PR: https://github.com/kubernetes/release/pull/1545)

**Which issue(s) this PR fixes**:

Fixes #
Tracking issue: https://github.com/kubernetes/kubernetes/issues/90698

**Special notes for your reviewer**:

Continuation of https://github.com/kubernetes/kubernetes/pull/90665 and https://github.com/kubernetes/kubernetes/pull/90697.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/assign @dims @BenTheElder 
cc: @kubernetes/release-engineering 
/priority important-soon

I'm expecting failures here as there are several places where it assumes that `pause` is going to be in the same image repository (`k8s.gcr.io`) as rest of the server images.

I've tracked down a bunch of those places, but I'm using the tests to catch the rest.

**Comment by dims** [comment]
Is this temporary? Just until the vanity domain flip happens?

**Comment by dims** [comment]
Can we please pause on this change? (pun intended)

**Comment by justaugustus** [comment]
> Is this temporary? Just until the vanity domain flip happens?
> Can we please pause on this change? (pun intended)

@dims -- 😂 Yep! @BenTheElder wasn't kidding when he said this was going to be a messy change.
There are a few places where it's basically a search/replace, but the kubeadm API change is likely verboten, and there are several places where I intentionally didn't bump the image (because they were using much older versions of `pause` and are probably candidates for deletion).

I'll sync with you and Ben on this early in the week and also get an updated timeline on the domain flip from @listx.

**Comment by justaugustus** [comment]
ref on the kubeadm issues: https://github.com/kubernetes/kubeadm/issues/2020

**Comment by dims** [comment]
💯 thanks @justaugustus

File: cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go [hunk 1]
```diff
@@ -101,6 +101,12 @@ type ClusterConfiguration struct {
 	// will be used for all the other images.
 	ImageRepository string `json:"imageRepository,omitempty"`
 
+	// PauseImageRepository sets the container registry to pull images from.
+	// If empty, `k8s.gcr.io` will be used by default; in case of kubernetes version is a CI build (kubernetes version starts with `ci/` or `ci-cross/`)
+	// `gcr.io/kubernetes-ci-images` will be used as a default for control plane components and for kube-proxy, while `k8s.gcr.io`
+	// will be used for all the other images.
+	PauseImageRepository string `json:"pauseImageRepository,omitempty"`
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go:108]: v1beta1 and 2 are frozen at this point.

**Comment by dims** [comment]
@justaugustus do we want to do this for 1.19? looks like we need a rebase at least

**Comment by justaugustus** [comment]
Changing direction here to instead build the pause image in k/release: https://github.com/kubernetes/release/pull/1545

/assign @dims 
cc: @kubernetes/release-engineering

**Comment by liggitt** [comment]
moving the pause image code out of k/k seems strange to me

**Comment by dims** [comment]
`pause` i feel is integral part of k/k, even though it has not changed much. go-runner was definitely a helper. i know ... the lines are not clear. probably better to stash this for now.

**Comment by BenTheElder** [comment]
+1 on moving the pause image code out of k/k being strange. This isn't a base / dependency image it's a critical component image.

**Comment by justaugustus** [comment]
Thanks for the feedback, everyone.
I've already closed the other related PRs yesterday.

I'll update this one later.

**Comment by sayanchowdhury** [comment]
👋🏽 Hey! I'm from the Bug Triage team. This PR has not been updated for a while, so I'd like to check on the status. The code freeze is starting November 12th (about 3 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.20, is it still planned for this release?

**Comment by justaugustus** [comment]
Forgot to close this out.
/close

**Pull request closed without merging**